## Microbial\_Metagenomics\_Workshop\_UCLA\_2015

##


This repository contains presentations, practicals and data files to be used. 

Specifically it contains:

1. This `README.md`
2. The Schedule of the course, `Schedule.md`
3. Daily repositories, `Monday` - `Friday`. 
4. LICENSE

##In Monday
1. A folder of the `Slides` for the day.
2. A folder for the `Command_line_practical` containing the instructions and data needed to complete the `exercises`.

##In Tuesday
1. A folder of the `Slides` for the day.
2. Summarize Taxa (QIIME) data files for each group (CS,LMJ,SSJ).

##In Friday
1. A folder of the `Slides` for the day.
2. `Metagenome predictions` for all groups (CS,LMJ,SSJ).
3. PiCrust/`Metagenome precictions`folder contains: all of the metagenome predictions in table format for each group. Use the files in exactly the same way the the taxa summary L6 files were used (pivot table, then primer). The files here show relative abundance. We realize that there are issues with this approach to metagenome analysis but those issues are not restricted to predicted function. We can discuss these issues further.
	- In addition to the KEGG pathway summary files (e.g. `Moorea_2015/All_Groups_metagenome_predictions/All_function_output/All_CbF_L3.txt`) the folders contain some plots and figures which each group can explore. Each folder has a `README` file included. 		
4. A folder containing additional data for group `SSJ` in: `SSJ_Cardiobacteriaceae_project`